### PR TITLE
Extend influxdb client to support changing of precision and measurements

### PR DESCRIPTION
**Pull request by royalaid**
Influxdb client can now take in :precision and :measurement  for the
0.9 client when those parameters need to be changed. It will act the
same as before when they are not supplied.

### CODE REVIEW

File: src/riemann/influxdb.clj [hunk 1]
```diff
@@ -151,13 +158,11 @@
   `:tag-fields`     A set of event fields to map into InfluxDB series tags.
                     (default: `#{:host}`)
   `:tags`           A common map of tags to apply to all points. (optional)
-  `:timeout`        HTTP timeout in milliseconds. (default: `5000`)"
+  `:timeout`        HTTP timeout in milliseconds. (default: `5000`)
+  `:measurement`    The name of the measurement to be used by influx
```
Comment (aphyr) [inline src/riemann/influxdb.clj]: The docs indicate :measurement is an argument to the client but it looks like instead you're using :measurement from the *event*? I'm a little confused about what's going on here.

Comment (royalaid) [inline src/riemann/influxdb.clj]: The :measurement is similar to series from influx 0.8 and it should be on a per event basis. Should I rework the docs?

File: src/riemann/influxdb.clj [hunk 2]
```diff
@@ -118,12 +118,19 @@
...
+            (str "&rp=" (:retention opts))))) 
+
 (defn event->point-9
   "Converts a Riemann event into an InfluxDB point if it has a time, service,
   and metric."
   [tag-fields event]
-  (when (and (:time event) (:service event) (:metric event))
-    {"measurement" (:service event)
+  (when (and (:time event) (or (:measurement event) (:service event)) (:metric event))
+    {"measurement" (:measurement event (:service event))
```
Comment (pyr) [inline src/riemann/influxdb.clj]: To make it more evident what you are doing here, please use:

```clojure
(or (:measurement event) (:service event))
```

File: src/riemann/influxdb.clj [hunk 3]
```diff
@@ -151,13 +158,11 @@
   `:tag-fields`     A set of event fields to map into InfluxDB series tags.
                     (default: `#{:host}`)
   `:tags`           A common map of tags to apply to all points. (optional)
-  `:timeout`        HTTP timeout in milliseconds. (default: `5000`)"
+  `:timeout`        HTTP timeout in milliseconds. (default: `5000`)
+  `:measurement`    The name of the measurement to be used by influx
+  `:precision`      The precision (n,u,ms,s,m,h) of the timestamp to be sent to influxdb (default: `s`)"
```
Comment (pyr) [inline src/riemann/influxdb.clj]: Why allow precision to be specified if there are no facilities to convert seconds to the other precisions?

**Comment by jamtur01** [comment]
@royalaid Any update on the feedback? Thanks!

File: src/riemann/influxdb.clj (hunk 3 above)
Comment (royalaid) [inline src/riemann/influxdb.clj]: Merely exposes the influxdb option so if you want to scale it for some reason you can.

**Comment by jamtur01** [comment]
Closing as stale. Please re-open if you get a chance! Thanks!

**Pull request closed without merging**